src/docker/workflows/confluent-local.ts

```diff
@@ -0,0 +1,339 @@
+import {
+  CancellationToken,
+  InputBoxValidationMessage,
+  InputBoxValidationSeverity,
+  Progress,
+  window,
+  workspace,
+  WorkspaceConfiguration,
+} from "vscode";
+import { findFreePort } from ".";
+import {
+  ContainerCreateRequest,
+  ContainerCreateResponse,
+  ContainerInspectResponse,
+  ContainerListRequest,
+  ContainerSummary,
+  HostConfig,
+} from "../../clients/docker";
+import { LOCAL_KAFKA_REST_PORT } from "../../constants";
+import { localKafkaConnected } from "../../emitters";
+import { Logger } from "../../logging";
+import { LOCAL_KAFKA_REST_HOST } from "../../preferences/constants";
+import { getLocalKafkaImageTag } from "../configs";
+import { MANAGED_CONTAINER_LABEL } from "../constants";
+import { createContainer, getContainersForImage } from "../containers";
+import { createNetwork } from "../networks";
+import { LocalResourceContainer, LocalResourceWorkflow } from "./base";
+
+export const CONTAINER_NAME_PREFIX = "vscode-confluent-local-broker";
+
+export class ConfluentLocalWorkflow extends LocalResourceWorkflow {
+  resourceKind: string = "Kafka";
+  static imageRepo = "confluentinc/confluent-local";
+
+  logger = new Logger("docker.workflow.confluent-local");
+
+  // ensure only one instance of this workflow can run at a time
+  private static instance: ConfluentLocalWorkflow;
+  private constructor() {
+    super(); // must call super() in derived class constructor
+  }
+  static getInstance(): ConfluentLocalWorkflow {
+    if (!ConfluentLocalWorkflow.instance) {
+      ConfluentLocalWorkflow.instance = new ConfluentLocalWorkflow();
+    }
+    return ConfluentLocalWorkflow.instance;
+  }
+
+  /**
+   * Start `confluent-local` resources locally:
+   * - Ensure the Docker image is available
+   * - Check for existing containers and exit early if found
+   * - Create a network if it doesn't exist
+   * - Create the container(s) for the Kafka broker(s)
+   * - Start the container(s)
+   * - Wait for the container(s) to be ready
+   */
+  async start(
```

Comment:
I'd like to see this God method broken even further into smaller side-methods, like the section for getting the number of brokers to start, etc. Have this evolve more into calling into each substep, then consulting the cancellation token and bailing if need be. The closer this reads to being like the docstring, the better, and variables declared only to carry info from a prior stage to a later one, the better.

Can come in future branch, though. This certainly works.